var/spack/repos/builtin/packages/m4/package.py

```diff
@@ -39,6 +39,16 @@ class M4(Package):
 
     depends_on('libsigsegv', when='+sigsegv')
 
+    def _make_executable(self, name):
+        return Executable(join_path(self.prefix.bin, name))
+
+    def setup_dependent_package(self, module, dependent_spec):
+        # m4 is very likely to be a build dependency, so we add the
```

Comment:
I added this because `autoconf` does the same for its scripts. I copied the code and the explanation from there.

`m4` is only an indirect dependency. Package `X` will depend on `autoconf` (build-time), and `autoconf` will depend on `m4` (run-time). Does this suffice?